Dashboard 1.6.2 crashes on Node view

##### Steps to reproduce
Deploy Dashbaord v1.6.2 and click on "Nodes"


##### Environment
```
Installation method: kubeadm
Kubernetes version: 1.7.2
Dashboard version: v1.6.2
Commit: b7669ed2081e7b444f8db000704bd7d7f61395b3
```


##### Observed result
Dashboard reported Unknown Server Error:
```
No error data available
```

```
[2017-07-28T07:27:34Z] Incoming HTTP/1.1 GET /api/v1/node?filterBy=&itemsPerPage=15&namespace=&page=1&sortBy=d,creationTimestamp request from 10.32.0.1:64136: {}
panic: runtime error: index out of range
goroutine 164 [running]:
panic(0x1c6d8e0, 0xc420014060)
	/home/travis/.gimme/versions/go1.7.1.linux.amd64/src/runtime/panic.go:500 +0x1a1
github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster.heapsterClient.ithResourceDownload.func1(0x2cc5e60, 0xc420392390, 0x1e95537, 0x4, 0x1e97a5e, 0x6, 0xc420b9a240, 0x2, 0x2, 0xc420a93e90, ...)
	/home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster/client.go:171 +0x5a5
created by github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster.heapsterClient.ithResourceDownload
	/home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster/client.go:177 +0x109
```

##### Comment
That only happens with the heapster fix from 
https://github.com/kubernetes/dashboard/pull/2181/files

Without the fix i can click on Node without any problem.

--- a/src/app/backend/integration/metric/heapster/client.go
+++ b/src/app/backend/integration/metric/heapster/client.go
@@ -162,21 +162,16 @@ func (self heapsterClient) ithResourceDownload(selector heapsterSelector, metric
 		}
 		dataPoints := DataPointsFromMetricJSONFormat(rawResult)
 
-		metric := &metricapi.Metric{
+		result.Metric <- &metricapi.Metric{
 			DataPoints:   dataPoints,
 			MetricPoints: toMetricPoints(rawResult.Metrics),
 			MetricName:   metricName,
-		}
-
-		if len(selector.Label[selector.TargetResourceType]) > i {
-			metric.Label = metricapi.Label{
+			Label: metricapi.Label{
 				selector.TargetResourceType: []types.UID{
 					selector.Label[selector.TargetResourceType][i],
 				},
-			}
+			},
 		}
-
-		result.Metric <- metric
 		result.Error <- nil
 		return
 	}()
@@ -205,21 +200,16 @@ func (self heapsterClient) allInOneDownload(selector heapsterSelector, metricNam
 		for i, rawResult := range rawResults.Items {
 			dataPoints := DataPointsFromMetricJSONFormat(rawResult)
 
-			metric := &metricapi.Metric{
+			result[i].Metric <- &metricapi.Metric{
 				DataPoints:   dataPoints,
 				MetricPoints: toMetricPoints(rawResult.Metrics),
 				MetricName:   metricName,
-			}
-
-			if len(selector.Label[selector.TargetResourceType]) > i {
-				metric.Label = metricapi.Label{
+				Label: metricapi.Label{
 					selector.TargetResourceType: []types.UID{
 						selector.Label[selector.TargetResourceType][i],
 					},
-				}
+				},
 			}
-
-			result[i].Metric <- metric
 			result[i].Error <- nil
 		}
 		return

--- a/src/app/backend/resource/node/common.go
+++ b/src/app/backend/resource/node/common.go
@@ -56,6 +56,7 @@ func (self NodeCell) GetResourceSelector() *metricapi.ResourceSelector {
 		Namespace:    self.ObjectMeta.Namespace,
 		ResourceType: api.ResourceKindNode,
 		ResourceName: self.ObjectMeta.Name,
+		UID:          self.ObjectMeta.UID,
 	}
 }
 

--- a/src/app/frontend/chrome/nav/nav.html
+++ b/src/app/frontend/chrome/nav/nav.html
@@ -35,8 +35,8 @@ limitations under the License.
   <div class="kd-nav-group-divider"></div>
   <kd-namespace-select></kd-namespace-select>
   <div class="kd-nav-group">
-      <kd-nav-item class="kd-nav-group-item"
-                   state="{{::$ctrl.states.overview}}">[[Overview|All objects group menu entry.]]</kd-nav-item>
+    <kd-nav-item class="kd-nav-group-item"
+                 state="{{::$ctrl.states.overview}}">[[Overview|All objects group menu entry.]]</kd-nav-item>
   </div>
   <div class="kd-nav-group">
     <kd-nav-item class="kd-nav-group-item"

--- a/src/app/frontend/overview/overview.html
+++ b/src/app/frontend/overview/overview.html
@@ -87,7 +87,7 @@ limitations under the License.
       </kd-stateful-set-card-list>
     </kd-content>
   </kd-content-card>
-  
+
   <kd-content-card ng-if="$ctrl.overview.ingressList.items.length">
     <kd-content>
       <kd-ingress-card-list ingress-list="$ctrl.overview.ingressList"

